test/client.js

```diff
@@ -348,6 +443,64 @@ function makeRequestAndExpectUrl (client, i, t, cb) {
   })
 }
 
+test('20 times HEAD with pipelining 10', (t) => {
+  const num = 20
+  t.plan(3 * num + 1)
+
+  let count = 0
+  let countGreaterThanOne = false
+  const server = createServer((req, res) => {
+    count++
+    setTimeout(function () {
+      countGreaterThanOne = countGreaterThanOne || count > 1
+      res.end(req.url)
+    }, 10)
```

Comment:
I think the problem is related to the timeout, which might happen before receiving more requests.

I would change this test to _not_ end the request until you have seen more than one parallel requests.

Reply:
You mean something like:
```js
    const interval = setInterval(() => {
      countGreaterThanOne = countGreaterThanOne || count > 1
      if (countGreaterThanOne) {
        clearInterval(interval)
        res.end(req.url)
      }
    }, 50)
```
Because neither this is working. It actually fails even if I disable all the checks and the count logic. The error is always 
![Screen Shot 2020-01-13 at 18 38 07](https://user-images.githubusercontent.com/15520377/72278238-dd2a7280-3633-11ea-865c-64fb037b0270.png)

I need to investigate further... Now I have no clue 😢

Reply:
I mean to _not rely on timers, intervals, etc_ at all. Rely on the sequence of events. You want to receive _10_ requests in parallel... wait to receive 10 requests before responding to them all.

Reply:
I investigated a bit further and I don't think it is something correlated to use of timers.
First, [the example](https://github.com/mcollina/undici/pull/9#discussion_r365336897) I posted before was wrong. Sorry about that. The client was not being initialized with pipelining.
I am much more inclined to think I am causing an issue with pipelining because using the server in `benchmark/origin.js` to test even something simple like
```js
const client = new Client('http://localhost:3000', {
    pipelining: 2
})

for (let i = 0; i < 4; i++) {
    client.request({
        path: '/',
        method: 'HEAD',
    }, (error, data) => {
        console.log(i)
        console.error(error)
        console.log(data)
        console.log()
    })
}
```
causes the last `n` requests to fail with the error `Error: other side closed - ended` from `undici/lib/client.js:47:23`. In this case is `n = 1` (the last request fails). If I try `100` requests with a pipelining factor of `10` the last `9` requests fail with the same error and so on...

This is the output of the server in debug mode responding to  4 HEAD requests with pipelining 2

![Screen Shot 2020-01-16 at 17 47 37](https://user-images.githubusercontent.com/15520377/72544909-7225a980-3888-11ea-9a42-f4fed1924617.png)

While this is 4 GET requests with pipelining 2

![Screen Shot 2020-01-16 at 17 47 01](https://user-images.githubusercontent.com/15520377/72544996-971a1c80-3888-11ea-8997-55307b5f2fdc.png)

I did try to remove the use of timers in the test, but I think the problem relies in my changes.

Reply:
is this still true? CI seems to be passing